### DISCUSSION TITLE
Hydrating custom objects using MongoDB with Panache

### DISCUSSION THREAD

**Discussion by UbiquitousBear** [quarkusio/quarkus#44049] [Q&A] [answered]
My entities have a custom object (used as an external identifier). The creation of the custom object needs request level information (request context) such as some sort of tenant or partition id. 

Is it possible to handle the persist using the repository pattern such that when Documents are converted to POJOs, the specific property/field is correctly hydrated? The property is already hydrated using a custom codec.

[a comment by an automated account is not shown]

**Reply by loicmathieu** [reply]
Normally, custom codecs are automatically detected and should be used, so it should already work.
Or maybe I missing something here.

**Reply by UbiquitousBear** [reply, marked as answer]
@loicmathieu Indeed; I'm using a custom codec to build an entity, but I'm going to close this discussion as I'm incorrectly approaching it
